pkg/controller/openliberty/libertyapplication_controller.go

```diff
@@ -346,6 +346,11 @@ func (r *ReconcileOpenLiberty) Reconcile(request reconcile.Request) (reconcile.R
 			autils.CustomizeStatefulSet(statefulSet, instance)
 			autils.CustomizePodSpec(&statefulSet.Spec.Template, instance)
 			autils.CustomizePersistence(statefulSet, instance)
+			if instance.Spec.CreateAppDefinition != nil && *instance.Spec.CreateAppDefinition {
```

Comment:
@JusteenR Users don't have to set `CreateAppDefinition` explicitly. It should be assumed to be true by default. Please check if it's nil or true. Thanks.

Reply:
Changed to account for the default.